storage/google/cloud/storage/blob.py

```diff
@@ -387,16 +432,25 @@ def generate_signed_url(
             client = self._require_client(client)
             credentials = client._credentials
 
-        return generate_signed_url(
+        if version == "v2":
```

Comment:
@tseaver could you add a warning as log output that V2 will change to V4 in the future:
>You have generated a signed URL using the default v2 signing implementation. In the future, this will default to v4. You may experience breaking changes if you use longer than 7 day expiration times with v4. To opt-in to the behavior specify `version="v2"`.

Reply:
a384692